crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -652,9 +662,19 @@ impl<'t> TextEdit<'t> {
             if text.as_str().is_empty() && !hint_text.is_empty() {
                 let hint_text_color = ui.visuals().weak_text_color();
                 let galley = if multiline {
```

Comment:
```suggestion
                let hint_text_font_id = hint_text_font.unwrap_or(font_id.into());
                let galley = if multiline {
```

to reduce code duplication a bit